### ISSUE TITLE
hightlight error with clang 3.8

### ISSUE THREAD

**Issue by Arondight** [jeaye/color_coded#118]
## with clang 3.8

![ ](http://i66.tinypic.com/2ym7ksy.png)

output for `:CCerror` is

```
/home/qinfd/.vim/bundle/color_coded/src/../include/env/environment.hpp:12:10: fatal error: 'env/impl.hpp' file not found
```

and i look up for this header file then get

```
» pkgfile impl.hpp
extra/boost
» pacman -Ql boost | grep -P '\/impl.hpp$'
boost /usr/include/boost/proto/transform/impl.hpp
boost /usr/include/boost/test/tools/old/impl.hpp
```

## without color_coded

```diff
--- a/vim/.vimrc
+++ b/vim/.vimrc
@@ -31,7 +31,7 @@ Plugin 'gmarik/Vundle.vim'
 "Plugin 'hdima/python-syntax'
 Plugin 'honza/vim-snippets'
 Plugin 'itchyny/calendar.vim'
-Plugin 'jeaye/color_coded'
+"Plugin 'jeaye/color_coded'
 Plugin 'jlanzarotta/bufexplorer'
 Plugin 'kien/rainbow_parentheses.vim'
 Plugin 'lilydjwg/colorizer'
```

and everything goes well.

## with clang 3.6

![ ](http://i68.tinypic.com/xnvok0.png)

## what i have done for new version

```bash
cd ~/.vim/bundle/color_coded
git pull
mkdir -p build
cd build
cmake ..
make install -j4
```

## make log

```bash
make install -j4 2>/dev/null
```

```
[  0%] Built target color_coded_track_api
[  7%] Building CXX object CMakeFiles/color_coded_boost.dir/lib/boost/filesystem/src/codecvt_error_category.cpp.o
[ 15%] Building CXX object CMakeFiles/color_coded_boost.dir/lib/boost/filesystem/src/operations.cpp.o
[ 23%] Building CXX object CMakeFiles/color_coded_boost.dir/lib/boost/filesystem/src/path_traits.cpp.o
[ 30%] Building CXX object CMakeFiles/color_coded_boost.dir/lib/boost/filesystem/src/path.cpp.o
[ 38%] Building CXX object CMakeFiles/color_coded_boost.dir/lib/boost/filesystem/src/portability.cpp.o
[ 46%] Building CXX object CMakeFiles/color_coded_boost.dir/lib/boost/filesystem/src/unique_path.cpp.o
[ 53%] Building CXX object CMakeFiles/color_coded_boost.dir/lib/boost/filesystem/src/utf8_codecvt_facet.cpp.o
[ 61%] Building CXX object CMakeFiles/color_coded_boost.dir/lib/boost/system/src/error_code.cpp.o
[ 69%] Linking CXX static library libcolor_coded_boost.a
[ 69%] Built target color_coded_boost
[ 76%] Building CXX object CMakeFiles/color_coded_config_test.dir/test/src/config/main.cpp.o
[ 84%] Building CXX object CMakeFiles/color_coded.dir/src/main.cpp.o
[ 92%] Linking CXX executable color_coded_config_test
[ 92%] Built target color_coded_config_test
[100%] Linking CXX shared library color_coded.so
[100%] Built target color_coded
Install the project...
-- Install configuration: "RELEASE"
-- Installing: /home/qinfd/.vim/bundle/color_coded/color_coded.so
-- Set runtime path of "/home/qinfd/.vim/bundle/color_coded/color_coded.so" to "/home/qinfd/.vim/bundle/color_coded/build/clang+llvm-3.8.0-x86_64-linux-gnu-ubuntu-14.04/lib"
```

## info of my system

```
» vim --version | grep -oP '[+-]lua'
+lua
» gcc --version | head -1
gcc (GCC) 5.3.0
» cat /etc/arch-release
Arch Linux release
» pacman -Q lua zlib clang ncurses xz
lua 5.3.2-1
zlib 1.2.8-4
clang 3.7.1-1
ncurses 6.0-4
xz 5.2.2-1
 » whereis lua.h 
lua: /usr/bin/lua /usr/bin/lua5.2 /usr/bin/lua5.3 /usr/include/lua.hpp /usr/include/lua.h /usr/include/lua5.2 /usr/share/man/man1/lua.1.gz
 » whereis libclang.so
libclang: /usr/lib/libclang.so
 » file /usr/lib/libtinfo.so
/usr/lib/libtinfo.so: symbolic link to /usr/lib/libncursesw.so.6
```

**Comment by Arondight** [comment]
this issue only effect c and cpp files

**Comment by Blimeo** [comment]
Exact same setup, same issue

**Comment by jeaye** [comment]
This is a duplicate of #120, I believe, which is a duplicate of #104. Both of those have the suggested fix to add to your .color_coded: `-fcolor-diagnostics`

**Comment by Arondight** [comment]
@jeaye It seems work on c files, but still has problem with cpp files.

```
» cat ~/.color_coded
-fcolor-diagnostics

```

**Comment by Arondight** [comment]
Sorry [now I kown why](https://github.com/jeaye/color_coded/blob/8a205973c5032c50266414f4caa7e8f7ea06739e/.color_coded), it only effect on plugin directory.

**Comment by Hexcles** [comment]
I believe we need some measures to address this `-fcolor-diagnostics` thing: either add a FAQ in README, or somehow preset this option inside color_coded.

**Comment by jeaye** [comment]
@Hexcles I've added this as a default and closed #104. Hopefully nobody experiences this any longer.

**Comment by pierremoreau** [comment]
@jeaye Thanks! This was particularly annoying when opening system headers. Maybe I should  have created a `.color_coded` in `/`…
